poem/Poem/frontend/react/UIElements.js

```diff
@@ -42,6 +52,15 @@ link_title.set('metrics', 'Metrics');
 link_title.set('metricprofiles', 'Metric profiles');
 link_title.set('aggregationprofiles', 'Aggregation profiles');
 
+export const link_icon = new Map();
+link_icon.set('administration', <FontAwesomeIcon icon={faWrench} fixedWidth/>);
+link_icon.set('services', <FontAwesomeIcon icon={faSitemap} fixedWidth/>);
+link_icon.set('serviceflavour', <FontAwesomeIcon icon={faHighlighter} fixedWidth/>);
+link_icon.set('reports', <FontAwesomeIcon icon={faFileAlt} fixedWidth/>);
+link_icon.set('probes', <FontAwesomeIcon icon={faServer} fixedWidth/>);
+link_icon.set('metrics', <FontAwesomeIcon icon={faCog} fixedWidth/>);
+link_icon.set('metricprofiles', <FontAwesomeIcon icon={faCogs} fixedWidth/>);
+link_icon.set('aggregationprofiles', <FontAwesomeIcon icon={faTasks} fixedWidth/>);
```

Comment:
This should be refactored in new stateless React component. Let the mapping structure be part of the component and based on the passed props, appropriate FontAwesomeIcon component should be returned. So, throughout the code, component should used something like this:
```
<Icon i="probes"/>
<Icon i="services"/>
...
```

Reply:
done